.circleci/config.yml

```diff
@@ -15,39 +15,22 @@ jobs:
             pip install \
               docker-compose==1.12.0 \
               awscli==1.11.76
-      - restore_cache:
-          keys:
-            - v1-{{ .Branch }}
-          paths:
-            - /caches/app.tar
       - run:
-          name: Load Docker image layer cache
-          command: |
-            set +o pipefail
-            docker load -i /caches/app.tar | true
-      - run:
-          name: Build application Docker image
-          command: |
-            docker build --cache-from=app -t app .
-      - run:
-          name: Save Docker image layer cache
-          command: |
-            mkdir -p /caches
-            docker save -o /caches/app.tar app
-      - save_cache:
-          key: v1-{{ .Branch }}-{{ epoch }}
-          paths:
-            - /caches/app.tar
+          name: Docker Build
+          command:
+            docker build .
       - run:
           name: Run tests
           command: |
             docker-compose -f ./docker-compose.test.yml up
       - deploy:
           name: Push application Docker image
           command: |
-            #if [ "${CIRCLE_BRANCH}" == "master" ]; then
+             if [ "${CIRCLE_BRANCH}" == "master" ]; then
               login="$(aws ecr get-login)"
               ${login}
-              docker tag app "${ECR_ENDPOINT}/${CIRCLE_PROJECT_REPONAME}2:${CIRCLE_PROJECT_REPONAME}${CIRCLE_SHA1}"
-              docker push "${ECR_ENDPOINT}/${CIRCLE_PROJECT_REPONAME}2:${CIRCLE_PROJECT_REPONAME}${CIRCLE_SHA1}"
-            #fi
\ No newline at end of file
+              docker tag app "${ECR_ENDPOINT}/${CIRCLE_PROJECT_REPONAME}:${CIRCLE_PROJECT_REPONAME}"
```

Comment:
@sablumiah ah, I think I was getting confused with the docker naming/tagging scheme because AWS use different terminology from all other Docker registries.

But, that current naming scheme means all containers exist in a flat namespace, so we'd have `laa-fee-calculator-web`, `formbuilder-admin` and the like all side-by-side. That would prevent us from writing IAM policies that restrict teams to have access to only their containers, so I'd suggest that we include the team name as well as repo name in the docker registry path - so something like this:

`${ECR_ENDPOINT}/${TEAM_NAME}/${CIRCLE_PROJECT_REPONAME}:${CIRCLE_SHA1}`

Then the IAM policy [over here](https://github.com/ministryofjustice/cloud-platform-reference-app/pull/33/files#diff-82037c6d6ceddad8191f453bdea61128R28) could be this, to grant a team access to all their containers, and their containers only:

`"Resource": "arn:aws:ecr:eu-west-1:${var.account_id}:${var.team_name}/*"`

Could maybe do with a second opinion on this though - thoughts @kalbir @ale-novo ?

Reply:
👍 I'd guess that teams will have multiple containers on the go almost immediately, so it's good to do it from the outset.

Also, we should be sure to standardise how team names are represented, as it'll crop up in various places across all aspects of the platforms. Also given that team names are already in use with Kubernetes authentication and AWS Federation, and are the slugified version of Github team names (which is the [final part of the URL when viewing a team](https://github.com/orgs/ministryofjustice/teams/opg-webops)), we should go with that across the board - I'll spread the word on that.